data/mods/gen3/random-teams.ts

```diff
@@ -371,43 +371,60 @@ export class RandomGen3Teams extends RandomGen4Teams {
 					cull = true;
 				}
 
-				const moveNeedsExtraChecks = (
+				const moveIsRejectable = (
 					!move.weather &&
 					(move.category !== 'Status' || !move.flags.heal) &&
 					(counter.setupType || !move.stallingMove) &&
+					// These moves cannot be rejected in favor of a forced move
 					!['batonpass', 'sleeptalk', 'solarbeam', 'substitute', 'sunnyday'].includes(moveid) &&
 					(move.category === 'Status' || !hasType[move.type] || (move.basePower && move.basePower < 40 && !move.multihit))
 				);
-				const otherMovesExtraChecks = (
+				// Pokemon should usually have at least one STAB move
+				const requiresStab = (
 					!counter.stab &&
-					!counter.damage &&
-					!counter.Ice &&
-					!counter.setupType &&
+					!hasMove['seismictoss'] && !hasMove['nightshade'] &&
+					species.id !== 'castform' &&
+					// If a Flying-type has Psychic, it doesn't need STAB
+					!(hasMove['psychic'] && hasType['Flying']) &&
+					!(hasType['Ghost'] && species.baseStats.spa > species.baseStats.atk) &&
+					!(
+						// With Calm Mind, Lugia and pure Normal-types are fine without STAB
+						counter.setupType === 'Special' &&
+						species.id === 'lugia' ||
+						(hasType['Normal'] && !species.types[1])
```

Comment:
```suggestion
						(hasType['Normal'] && species.types.length < 2)
```

Reply:
i was just being consistent with other parts of the file honestly but sure